SAMDP test failures

```
pip install -U scipy==1.7.2
Collecting scipy==1.7.2
  Using cached scipy-1.7.2-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (39.8 MB)
Requirement already satisfied: numpy<1.23.0,>=1.16.5 in /home/rene/.local/share/virtualenvs/main_pymor/lib/python3.9/site-packages (from scipy==1.7.2) (1.21.4)
Installing collected packages: scipy
  Attempting uninstall: scipy
    Found existing installation: scipy 1.7.1
    Uninstalling scipy-1.7.1:
      Successfully uninstalled scipy-1.7.1
Successfully installed scipy-1.7.2
WARNING: You are using pip version 21.2.4; however, version 21.3.1 is available.
You should consider upgrading via the '/home/rene/.local/share/virtualenvs/main_pymor/bin/python -m pip install --upgrade pip' command.
(main_pymor) ┐ (rene@bbeight) [09:14:14|Mi Dez 29] [tests_no_mess] [~/projekte/uni/pymor/working-trees/main_pymor] 
└───> pytest src/pymortests/algorithms/samdp.py 
============================================== test session starts ===============================================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /home/rene/projekte/uni/pymor/working-trees/main_pymor, configfile: setup.cfg
plugins: cov-3.0.0, xdist-2.4.0, forked-1.3.0, anyio-3.3.4, datadir-1.3.1, parallel-0.1.1, hypothesis-6.24.2
collected 96 items                                                                                               

src/pymortests/algorithms/samdp.py ....................................................................... [ 73%]
.........F.......F......

~snip~

============================================ short test summary info =============================================
FAILED src/pymortests/algorithms/samdp.py::test_samdp[True-NM-15-2-1-50] - pymor.core.exceptions.InversionError...
FAILED src/pymortests/algorithms/samdp.py::test_samdp[True-NM-20-2-1-50] - pymor.core.exceptions.InversionError...
========================================= 2 failed, 94 passed in 59.70s ==========================================

```

Happens with 1.7.2, 1.7.3 and 1.8.0rc2. Does not happen in the docker container for gitlab CI, but does happen locally in venv and in #1518 virtual machines with conda.

Would be great if somebody else from @pymor/all could look into this.

--- a/.ci/azure/pipeline-osx.yml
+++ b/.ci/azure/pipeline-osx.yml
@@ -29,7 +29,7 @@ jobs:
       #conda update -y --all
       conda install -y --only-deps pymor
       # these ones are not in the 2020.1 conda build yet
-      conda install -y pyevtk mpi4py slycot pytorch pyqt==5.12.3 pyside2==5.13.2 codecov qtpy
+      conda install -y pyevtk mpi4py slycot pytorch pyqt==5.12.3 pyside2==5.13.2 codecov "qtpy<2.0.0"
       # these are buildtime, not a runtime,  deps for our conda package
       conda install -y cython pytest-cov pytest
       # install anything which might be a new dependency with pip

--- a/.ci/azure/pipeline-win.yml
+++ b/.ci/azure/pipeline-win.yml
@@ -44,7 +44,7 @@ jobs:
     displayName: Conda 2/3
   - script: |
       call activate myEnvironment
-      conda install -v -c conda-forge --yes --name myEnvironment pyopengl cython pyevtk slycot cython pytest pytest-cov curl hypothesis pyside2==5.13.2 qtpy typer click==7.1.2
+      conda install -v -c conda-forge --yes --name myEnvironment pyopengl cython pyevtk slycot cython pytest pytest-cov curl hypothesis pyside2==5.13.2 "qtpy<2.0.0" typer click==7.1.2
     displayName: Conda 3/3
 
   - script: |

--- a/dependencies.py
+++ b/dependencies.py
@@ -39,7 +39,7 @@ def setup_requires():
 # recheck if jupyter_client pin still necessary
 #   https://github.com/jupyter-widgets/pythreejs/issues/366
 # Qt bindings selectors are a woraround for https://bugreports.qt.io/browse/QTBUG-88688
-install_requires = ['qtpy', 'packaging', 'diskcache', 'typer', 'click'] + _numpy_scipy()
+install_requires = ['qtpy!=2.0.0', 'packaging', 'diskcache', 'typer', 'click'] + _numpy_scipy()
 install_suggests = {
     'ipython>=5.0': 'an enhanced interactive python shell',
     'ipyparallel>=6.2.5': 'required for pymor.parallel.ipython',

--- a/requirements-optional.txt
+++ b/requirements-optional.txt
@@ -32,7 +32,7 @@ pyopengl
 pytest>=6.0
 python-slugify
 pythreejs
-qtpy
+qtpy!=2.0.0
 scipy>=1.3.3;python_version == "3.8"
 scipy>=1.3;python_version < "3.8"
 scipy>=1.5.4;python_version >= "3.9"

--- a/requirements.txt
+++ b/requirements.txt
@@ -6,7 +6,7 @@ numpy>=1.17.5;python_version == "3.8"
 numpy>=1.19.4;python_version >= "3.9"
 packaging
 pytest-runner>=2.9
-qtpy
+qtpy!=2.0.0
 scipy>=1.3.3;python_version == "3.8"
 scipy>=1.3;python_version < "3.8"
 scipy>=1.5.4;python_version >= "3.9"

--- a/src/pymor/algorithms/samdp.py
+++ b/src/pymor/algorithms/samdp.py
@@ -436,7 +436,7 @@ def _twosided_rqi(A, E, x, y, theta, init_res, imagtol, rqitol, maxiter):
         x = x_rqi
         y = v_rqi
         nrq = rqi_res.norm() / np.abs(x_rq)
-        if np.abs(theta - x_rq) / np.abs(x_rq) < rqitol:
+        if nrq < rqitol:
             break
         theta = x_rq
         if not np.isfinite(nrq):

